vmc/utils.go

```diff
@@ -90,14 +90,19 @@ func getNSXTReverseProxyURLConnector(nsxtReverseProxyUrl string) (client.Connect
 	return connector, nil
 }
 
-func getTotalSddcHosts(sddc *model.Sddc) int {
-	totalHosts := 0
-	if sddc != nil && sddc.ResourceConfig.Clusters != nil {
+// getHostCountCluster tries to find the amount of hosts on a Cluster in
+// the ResourceConfig of the provided SDDC. If there is no ResourceConfig/Cluster 0 is returned.
+// A Cluster is distinguished by its id
+func getHostCountCluster(sddc *model.Sddc, clusterId string) int {
+	clusterHostCount := 0
+	if sddc != nil && sddc.ResourceConfig != nil && sddc.ResourceConfig.Clusters != nil {
 		for _, cluster := range sddc.ResourceConfig.Clusters {
-			totalHosts += len(cluster.EsxHostList)
+			if cluster.ClusterId == clusterId {
+				clusterHostCount += len(cluster.EsxHostList)
```

Comment:
Could there be multiple clusters with the same ID in `sddc.ResourceConfig.Clusters`?
(I'm wondering why we don't return here instead of looping through)

Reply:
Shortsighted refactor on my part. Will fix on next MR.